tests/serializers/test_data.py

```diff
@@ -477,13 +432,76 @@ def serializerTest(self, format):
 
     # Assert that the deserialized data is the same
     # as the original source
-    for func, pk, klass, datum in test_data:
-        func[1](self, pk, klass, datum)
+    for func, pk, klass, datum in data:
+        with self.subTest(klass=klass, data=datum):
+            func[1](self, pk, klass, datum)
 
     # Assert that the number of objects deserialized is the
     # same as the number that was serialized.
     for klass, count in instance_count.items():
-        self.assertEqual(count, klass.objects.count())
+        with self.subTest(klass=klass, count=count):
+            self.assertEqual(count, klass.objects.count())
+
+
+def serializerTest(self, format):
+    assert_serializer(self, format, test_data)
+
+
+@skipUnlessDBFeature("allows_auto_pk_0")
+def serializerTestAllowPK0(self, format):
+    # FK to an object with PK of 0. This won't work on MySQL without the
+    # NO_AUTO_VALUE_ON_ZERO SQL mode since it won't let you create an object
+    # with an autoincrement primary key of 0.
+    data = [
+        (data_obj, 0, Anchor, "Anchor 0"),
+        (fk_obj, 465, FKData, 0),
+    ]
+    assert_serializer(self, format, data)
+
+
+@skipUnlessDBFeature("interprets_empty_strings_as_nulls")
+def serializerTestInterpretsEmptyStringsAsNulls(self, format):
+    # Because Oracle treats the empty string as NULL, Oracle is expected to fail
+    # when field.empty_strings_allowed is True and the value is None; skip these
+    # tests.
+    data = [
```

Comment:
1. Do we care that the PK sequence is kind of losing its meaning since these and others are being moved out of `test_data`? It doesn't matter practically, but I wonder if it's a bit confusing now.
2. Would there be any value in ordering these by PK? What ordering are you using here?

Reply:
I'd vote to reindex the pks (starting at 1, going by increments of 1) for the new test functions (`serializerTestAllowPK0`, `serializerTestInterpretsEmptyStringsAsNulls`, and `serializerTestSupportsIndexOnTextField`). I've tested it and it works for me locally (which is what I'd assumed since each test methods runs in a fresh db as far as I understand).

For the old `test_data` I'd keep the legacy pks to minimize the diff, maybe with a comment that explains why the pk sequence has gaps.

Reply:
So I think the pk sequence has little meaning with a couple of exceptions:
- `serializerTestAllowPK0` as it has a pk of 0
- any one where they reference eachother (see stuff with the comments `# Post reference` and `# Pre reference`)

The rest don't matter very much, they are used as a unique identifier (we probably could use the data itself as the unique identifier)